optuna/samplers/_nsga2/sampler.py

```diff
@@ -154,18 +271,39 @@ def sample_relative(
         generation = parent_generation + 1
         study._storage.set_trial_system_attr(trial_id, _GENERATION_KEY, generation)
 
+        dominates_func = _dominates if self._constraints_func is None else _constrained_dominates
+
         if parent_generation >= 0:
-            p0 = self._select_parent(study, parent_population)
+            # We choose a child based on the specified crossover method.
             if self._rng.rand() < self._crossover_prob:
-                p1 = self._select_parent(
-                    study, [t for t in parent_population if t._trial_id != p0._trial_id]
+                child_params = crossover(
+                    self._crossover,
+                    study,
+                    parent_population,
+                    search_space,
+                    self._rng,
+                    self._swapping_prob,
+                    dominates_func,
                 )
             else:
-                p1 = p0
+                parent_population_size = len(parent_population)
+                child_params = {}
+                for param_name in search_space.keys():
+                    child_params[param_name] = parent_population[
+                        self._rng.choice(parent_population_size)
+                    ].params[param_name]
```

Comment:
It might be fine but did you intentionally change the logic to use different parents for different genes/params?

Reply:
With the previous code, it was possible for the child to have keys that were not in `search_space.keys()`, which would lead to bugs.
Therefore, only the keys in `search_space.keys()` are retrieved from the parent.

However, as you said, with this code, the child may have different parameters from different parent individuals, so I have to improve the code as follows.
```suggestion
                parent_params = parent_population[self._rng.choice(parent_population_size)].params
                for param_name in search_space.keys():
                    child_params[param_name] = parent_params[param_name]
```

Reply:
The change looks correct. I think we should rerun the experiment although effects might be small (also given that the crossover probably used before was default 0.9/relative high?) though. Do you have time for this? Also, let me just correct my previous comment. The current logic in a sense does a crossover (although the code branch is for no crossover) so I think we should fix it.